docker/Dockerfile

```diff
@@ -1,26 +1,84 @@
+#  ultralytics, base
+ARG DEPS="all"
+ARG VERSION
+ARG BRANCH
 
-# Install git
-RUN : \
+RUN set -Eeuxo \
     && apt-get update \
+    && apt-get install ffmpeg libsm6 libxext6  -y \
     && DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends git \
-    && apt-get clean \
-    && rm -rf /var/lib/apt/lists/*
+    && python3.8 -m venv $VENV \
+    && $VENV/bin/pip install --no-cache-dir --upgrade pip setuptools wheel
 
-# Activate venv
-RUN python3.8 -m venv /venv
-ENV PATH="venv/bin:$PATH"
+RUN \
+    if [ -z "${VERSION}" ] ; then  \
+    echo VERSION not provided && \
+      if [ -z "${BRANCH}" ]; then  \
+        git clone https://github.com/neuralmagic/deepsparse.git --depth 1 -b main; \
+      else clone https://github.com/neuralmagic/deepsparse.git --depth 1 -b $BRANCH; \
+      fi; \
+    else echo VERSION is $VERSION &&  \
+      git clone https://github.com/neuralmagic/deepsparse.git --depth 1 -b release/${VERSION}; \
+    fi;
 
-RUN pip3 install --upgrade setuptools wheel
 
-# Setup DeepSparse
 
-ARG GIT_CHECKOUT
-# if $GIT_CHECKOUT is not specified - just install from pypi
-RUN if [ -z "${GIT_CHECKOUT}" ] ; then pip3 install --no-cache-dir --upgrade deepsparse[server] ; fi
+FROM base AS container_branch_all
+ARG VENV
+ENV PATH="${VENV}/bin:$PATH"
+ENV PIP_DEFAULT_TIMEOUT=200
+ARG DEEPSPARSE_COMMAND=./deepsparse[server,yolo,onnxruntime,yolov8]
+RUN $VENV/bin/pip install install --no-cache-dir "${DEEPSPARSE_COMMAND}"
 
-# if $GIT_CHECKOUT is specified - clone, checkout $GIT_CHECKOUT, and install with -e
-RUN if [ -n "${GIT_CHECKOUT}" ] ; then git clone https://github.com/neuralmagic/deepsparse.git --depth 1 -b $GIT_CHECKOUT; fi
-RUN if [ -n "${GIT_CHECKOUT}" ] ; then pip3 install --no-cache-dir --upgrade -e "./deepsparse[server]" ; fi
+RUN deepsparse.transformers.run_inference --help \
+    && deepsparse.image_classification.annotate --help
+
+
+FROM base AS container_branch_server
+ARG VENV
+ENV PATH="${VENV}/bin:$PATH"
+ENV PIP_DEFAULT_TIMEOUT=200
+ARG DEEPSPARSE_COMMAND=./deepsparse[server]
+RUN $VENV/bin/pip install install --no-cache-dir "${DEEPSPARSE_COMMAND}"
+
+FROM container_branch_server AS container_branch_transformers
+RUN deepsparse.transformers.run_inference --help
+
+
+FROM container_branch_server AS container_branch_torchvision
+RUN deepsparse.image_classification.annotate --help
+
+
+FROM base AS container_branch_ultralytics
+ARG VENV
+ENV PATH="${VENV}/bin:$PATH"
+ENV PIP_DEFAULT_TIMEOUT=200
+ARG DEEPSPARSE_COMMAND=./deepsparse[server,yolo,yolov8]
+RUN $VENV/bin/pip install install --no-cache-dir "${DEEPSPARSE_COMMAND}"
+
+
+FROM base AS container_branch_base
+ARG VENV
+ENV PATH="${VENV}/bin:$PATH"
+ENV PIP_DEFAULT_TIMEOUT=200
+ARG DEEPSPARSE_COMMAND=./deepsparse
+RUN $VENV/bin/pip install install --no-cache-dir "${DEEPSPARSE_COMMAND}"
+
+FROM base AS container_branch_dev
+ARG VENV
+ENV PATH="${VENV}/bin:$PATH"
+ENV PIP_DEFAULT_TIMEOUT=200
+ARG DEEPSPARSE_COMMAND=./deepsparse[dev]
```

Comment:
in developer mode we should be installing against a local copy of the repo, no? so either copying it from the users base machine or cloning it to allow for an editable install

Reply:
Great callout, I added the -e flag, however we do not need to do anything special for a dev container, we can get that by supplying a `--target` flag like the following:

`docker build --target container_branch_dev --build-arg DEPS=dev -t deepsparse-dev-docker . && docker container run -it deepsparse-dev-docker bash`